### ISSUE TITLE
TODO: Despawn NPCs with high NPC entry

### ISSUE THREAD

**Issue by Daribon** [The-Alpha-Project/alpha-core#313]
**There are vast amount of NPC entries at high entry that should not be spawned in alpha.**

When it comes to totems there are several NPC entries that does not exist in 0.5.3 spell dbc:
![image](https://user-images.githubusercontent.com/6137576/178376673-3ab7fb46-a5bd-4dc4-b09f-bc014fb1ac43.png)
![image](https://user-images.githubusercontent.com/6137576/178376684-d313c1a5-7f03-45cd-8c0a-8f65e3cbcf1e.png)
However the spell could have been added after the npc getting created.
Since not even Stoneskin totem has in the 0.5.3 dbc, but rank 1 npc has entry 5873

Last NPC totem entries that has spells in dbc:
![image](https://user-images.githubusercontent.com/6137576/178377125-b242f577-6863-4e06-a410-434bc1a373b3.png)

![heidisql_F2KzUY4zlE](https://user-images.githubusercontent.com/6137576/178381191-3ec328c8-8a26-4e17-9923-73e8d5f719ba.png)
![heidisql_0ljcvM1ZFc](https://user-images.githubusercontent.com/6137576/178381216-0b89bacf-602e-4f80-bc01-72fae89b2f24.png)
Between 5,9k and 6,1k tons of totems added not available in 0.5.3 but probably in 0.5.4.

![fevrier2004-01](https://user-images.githubusercontent.com/6137576/178556444-9523f406-85bb-421e-a7f9-5d419f4ffb9b.jpg)
Alyssa Blaye has entry 5665 and the image is from 0.5.4, fishing in UC slime wasn't possible until 0.5.4.

![EarlyEnchanting](https://user-images.githubusercontent.com/6137576/178548952-ec884be1-849e-4325-9db5-1af265adf953.jpg)
Vance Undergloom has entry 5695 and the image is from 0.5.3.

![unindel_wow55](https://user-images.githubusercontent.com/6137576/178548100-d8c410b7-dcfd-41e4-b3f9-ba8e7addaea2.JPG)
From 0.5.3. Winifred Kerwin has entry 5703. Image dated 2003-12-14.
The summoned NPCs exist as well at just a bit higher entry:
![image](https://user-images.githubusercontent.com/6137576/178548345-31bdbdc9-e485-445f-b39b-414e2624a1dd.png)

![image](https://user-images.githubusercontent.com/6137576/178377131-42156b3d-c393-4164-87a0-ead48919a207.png)
Up to entry 5700 100% confirmed for being in alpha.
Guardian of blizzard is located at entry 5764, NPC was used to prevent players from entering unfinished locations.
I've only been able to find images of Guardian of Blizzard from 0.5.4 and above.
Image from 0.5.4:
![shot8](https://user-images.githubusercontent.com/6137576/178561678-a2adfdaf-a3a3-4f31-9da8-11d2ef5951ae.jpg)
![heidisql_1T9gjGSqlt](https://user-images.githubusercontent.com/6137576/178567054-ece86762-725b-49e5-916a-68541a401423.png)
Guardian of Blizzard has entries above and below it which are from either WC or Barrens, fitting the 0.5.4 theme of Tri-Horde push (Troll, Orc, Tauren).
Going into broadcast text and looking at the text that Guardian of Blizzard uses one can see that its broadcast text entry is located almost right next to last entry for Undercity broadcast texts:
![EmEditor_GMBj0jEiCx](https://user-images.githubusercontent.com/6137576/178566562-55ca3333-56ab-463e-b402-7d31b353dd2b.png)
The very last entry next to Guardian of Blizzard text "%s whispers to $N the secret to opening his strongbox." is used by unknown NPC. Might be unused text used for the Apothecary Keever event, which the entry next to it belongs to.

However we can dig even deeper than this.
What are the last displayids used in 0.5.3?
![explorer_EYzWLZAipV](https://user-images.githubusercontent.com/6137576/178569937-bba294bb-214b-45e8-bc59-bb05cc1b7a67.png)
What displayid of those are used by an NPC with high NPC entry. Answer is displayid 4179.
The displayid 4179 is used by NPC entry 5759.
![heidisql_RS9q7jUF6O](https://user-images.githubusercontent.com/6137576/178570188-c454c29e-9f23-434c-9f0c-872e7b57d3ef.png)
The NPC entry next to 5759 uses a displayid not available in 0.5.3 and is located in kalimdor, fitting for 0.5.4. Following entries are from WC, also fitting for 0.5.4 Tri-horde push.

The last NPC entry in the 0.5.3 client should be 5759.

![image](https://user-images.githubusercontent.com/6137576/178377203-8e963275-bd4d-47d3-84df-4475ea4fc116.png)
Entry 6706 flight master in Nijhels point, the location has a gryphon stand in 0.5.3 but is very very unfinished. Nijhels point didn't receive its final look until 0.10.0.3892, with only a bridge and the location being retextured between 0.5.3 and 0.10.0. Nijhels point does not exist in TaxiNodes.dbc for 0.5.3, which means it was never any location where you could use flight path in 0.5.3. Nijhels flight path was added in 0.6.0.
![firefox_fJQsSoZVM8](https://user-images.githubusercontent.com/6137576/178384546-928f14b7-dd2b-43b7-81ef-84ff42c9f90b.png)
Entry 6726, flight master in Shadowprey Village, that village doesn't even exist in alpha. Shadowprey Village was added in 0.10.0.3892. 
If Shadowprey Village didn't exist then why does this entry exist at such a low entry?
Because the NPC wasn't spawned in Shadowprey Village, it used to be spawned in Ghost Walker Post in 0.6.0, when the flight path was added, from TaxiNodes.dbc 0.6.0: 
![firefox_djI2ATNkbL](https://user-images.githubusercontent.com/6137576/178388132-b8e65c6f-a6a8-4521-875e-029a04f401cc.png)

![firefox_3P2lEr8Xss](https://user-images.githubusercontent.com/6137576/178537048-78331686-f11d-4c67-a7f9-39d544019f2b.png)
Last entry of flight master NPC in the 0.5.3 which has a node in taxipathnodes.dbc is 4407 Teloren in Stonetalon Peak.
Not a single flightmaster entry after 4407 has any node in taxipathnodes.
The first flight master NPC entry that shows up after 4407 is 6026, Breyk, flightmaster in Swamp of Sorrows. The first time Swamp of Sorrows shows up in taxipathnode is 0.5.5, which means that it was either added in 0.5.4 or 0.5.5 (since we do not have the 0.5.4 we cannot check).

Then there is this:
![image](https://user-images.githubusercontent.com/6137576/178377459-0a75e7b5-fb80-4b80-a565-6937fe3b6600.png)
It continues with other entires with a few entries later down too.
With innkeepers come you guessed it, Spirit healer:
![image](https://user-images.githubusercontent.com/6137576/178377521-3b7dcb45-e0b4-4725-8332-92c0db77be61.png)

Ghost spirit healer were added in patch 0.6.
The tons of innkeepers getting added starting with 6,7k entry? Because hearthstones got added in 0.6.
![firefox_5yK3zFTT91](https://user-images.githubusercontent.com/6137576/178378618-c8bf905b-beaa-44a8-a7da-75e2fe1179ca.png)

![heidisql_791z5asyBr](https://user-images.githubusercontent.com/6137576/178380876-8e3a57db-0928-49ba-bac5-93739ee23e1f.png)
7k entries almost instantly start with winterspring NPCs, a zone completely unfinished in alpha.

![heidisql_YrdtxcH9ws](https://user-images.githubusercontent.com/6137576/178380976-7f76d144-3819-470e-8e8d-a63479439b62.png)
Undercity entries are at around 4,5k and knowing that it UC was added in 0.4 we can build on top of that.
![heidisql_vbPacYLnUh](https://user-images.githubusercontent.com/6137576/178381058-c1c0f59f-5b4d-4cb3-b39f-033112738332.png)
Knowing that ironforge wasn't finished a month prior to human push 0.5 (and might not even been finished in human push) it too is reflected in NPC entries, around Ironforge entries are at around early 5k.

![heidisql_mLd2Ilu69Z](https://user-images.githubusercontent.com/6137576/178382404-dc6015c0-b9b1-49c5-b06b-0b1385670379.png)
![heidisql_tbY03zDklo](https://user-images.githubusercontent.com/6137576/178381790-a8df84c8-b339-439e-bb51-021e3b2ff8f0.png)
7314, TEST battlemaster and a bunch of racial mounts added between at around 7,6k.
Racial mounts was added in 0.7.0 and battlegrounds was announced in 0.7.0, however it was not available on the beta.
![image](https://user-images.githubusercontent.com/6137576/178377851-afc5eeb5-5c40-4793-8c56-77d6479a3bd5.png)
![image](https://user-images.githubusercontent.com/6137576/178377858-319e88ec-026d-4ccb-8459-270cdb6f6bcf.png)

Patch 0.8 added auction house.
![image](https://user-images.githubusercontent.com/6137576/178377740-c7a7bd59-fae7-409f-86e9-33fc4258bb5f.png)
And the auctioneer NPCs appear about 1k entries above known what is presumed to be 0.7.0 entries.

It appears that blizz added about 500 NPC entries per patch.

![heidisql_agbSboXvr0](https://user-images.githubusercontent.com/6137576/178383540-4266c2eb-f209-40a9-b296-c425d765927d.png)
Entry 10919, Shatterspear Troll in Shatterspear village.
The entire village didn't exist until 0.10.0.3892.

![CG6dT1TUkAAVJkN](https://user-images.githubusercontent.com/6137576/178529588-abe909f5-2064-44be-a9ef-90de0d841c83.png)
Onyxia, entry 10184 was first spawned on August 12, 2004. It is possible that the entry was created earlier but considering raid content was rushed most likely not much earlier.

![heidisql_SyQR7OmZox](https://user-images.githubusercontent.com/6137576/178383467-c010975a-c74e-4f81-b28b-fe684c5cc0a7.png)
Sylvanas wasn't added until at least 0.9.0+ and her entry is 10181.
![WoWScrnShot_052004_222822](https://user-images.githubusercontent.com/6137576/178532619-07b1d8b2-412a-4601-b0bf-40c88b22a3c5.jpg)
Image from 20-05-2004 still does not have her spawned.
![62](https://user-images.githubusercontent.com/6137576/178533021-1cc72905-01d6-44c0-b45a-19049be0dfe3.jpg)
First image with Sylvanas showing uses a placeholder model and is from patch 0.9.0, or above (due to talent icon being 0.9.0+.

Starting NPCs in Eversong Woods in TBC are at around entry 15k.
Those entries were added sometime between early 2005 and blizzcon 2005 (october 2005)
Molten Core entries start at about 12k, which was developed at very end before wow launch.
Meanwhile last entry of vanilla is 18k.
It is therefor extremely unlikely that entries above 10k should exist in 0.5.3.

TL;DR only entries up to 5700 are 100% confirmed have existed during 0.5.3.

**Describe what needs to be implemented or enhanced**

Despawn all entries above 10k
UPDATE `spawns_creatures` SET `ignored` = 1 WHERE `spawn_entry1` > 10000;

I do not recommend going lower yet, due to the large amount of NPCs that get despawned. It has to be researched more before going lower and double checked that no entries that possibly should be in alpha, do not get despawned.

**Expected end result**
Clears up NPC spawns that should clearly not be in alpha.
Considering only entries above 10k is despawned thus far, not much should change.
Thus far I've only seen silithus completely despawn, which is expected, considering how it was unfinished even at release.

**Comment by Daribon** [comment]
I've done tons of digging today and from my research, the last NPC entry used in the 0.5.3 should be 5759.
The findings are documented under the part which begins with Guardian of Blizzard.

**Comment by GrenderG** [comment]
Amazing research. :+1:
